Training24/Test.c

```diff
@@ -1,19 +1,187 @@
+}
+
+void TestPhrasePalindromes () {
+   PrintTableHeader (Phrase);
+   const char* phrases[] = { "tacOcAT", "hello", "D", "Don't nod", "Race!Car", " ", "22 / 02 / 2022",
+      "12345678987654321", "0123456789" },
+      * expectedResults[] = { "Palindrome", "Not a Palindrome", "Palindrome", "Palindrome",
+      "Palindrome", "Palindrome", "Palindrome", "Palindrome", "Not a Palindrome" };
+   for (int i = 0; i < sizeof (phrases) / sizeof (phrases[0]); i++) {
+      int isPhrasePalindrome = CheckPhrasePalindrome (phrases[i]);
+      const char* actualResult = isPhrasePalindrome ? "Palindrome" : "Not a Palindrome",
+         * output = (strcmp (actualResult, expectedResults[i]) == 0) ? GREEN "PASS" RESET : RED "FAIL" RESET;
+      PhraseTableRow (phrases[i], expectedResults[i], actualResult, output);
+   }
+   printf ("\n");
+}
+
+void TestReverseNumbers () {
+   int testCases[] = { 212, -121, 12345, -9876789, 999999999, 2147447412, 2147483647, -2147483648LL }, reversedNumber;
+   const char* expectedOutputs[] = { "Palindrome", "Not a Palindrome", "Not a Palindrome",
+      "Not a Palindrome", "Palindrome", "Palindrome", "Failed to process", "Not a Palindrome" };
+   PrintTableHeader (Integer);
+   for (int i = 0; i < sizeof (testCases) / sizeof (testCases[0]); i++) {
+      int number = testCases[i];
+      char outputStr[256];
+      bool overflow = false;
+      if (number < 0) IntegerTableRow (number, "Invalid(Negative number)", "Not a Palindrome", expectedOutputs[i]);
+      else {
+         overflow = ReverseAndCheckOverflow (number, &reversedNumber);
+         if (overflow) {
+            strcpy (outputStr, "Integer Overflow");
+            IntegerTableRow (number, outputStr, "Failed to process", expectedOutputs[i]);
+         }
+         else {
+            sprintf (outputStr, "%d", reversedNumber);
+            const char* actualOutput = IsNumPalindrome (number) ? "Palindrome" : "Not a Palindrome";
+            IntegerTableRow (number, outputStr, actualOutput, expectedOutputs[i]);
+         }
+      }
+   }
+}
+
+void TestUserPhrase () {
+   char userPhrase[500];
+   while (1) {
+      printf ("\nEnter a phrase to test (max 499 characters, or type 'r' to return menu): ");
+      if (fgets (userPhrase, sizeof (userPhrase), stdin) == NULL) {
+         printf (YELLOW "Error: Input not received. Exiting function." RESET "\n");
+         return;
+      }
+      userPhrase[strcspn (userPhrase, "\n")] = 0;
+      if (strcmp (userPhrase, "r") == 0 || strcmp (userPhrase, "R") == 0) return;   // Return to menu command
+      if (strlen (userPhrase) >= sizeof (userPhrase) - 1) {
+         printf (YELLOW "Error: Input exceeds maximum length." RESET "\n");
+         while (getchar () != '\n' && getchar () != EOF);
+         continue;
+      }
+      else printf ("Result: %s\n", CheckPhrasePalindrome (userPhrase) ? GREEN "It's a Palindrome" RESET : RED "Not a Palindrome" RESET);
+   }
+}
+
+void TestUserInteger () {
```

Comment:
You don't test user input (there is no point). The point of testing is to test your logic (for yourself alone) so you know that the code you're sending to the user has no bugs.